evm/src/main/java/org/hyperledger/besu/evm/gascalculator/PragueGasCalculator.java

```diff
@@ -68,4 +73,20 @@ public long delegateCodeGasCost(final int delegateCodeListLength) {
   public long calculateDelegateCodeGasRefund(final long alreadyExistingAccounts) {
     return existingAccountGasRefund * alreadyExistingAccounts;
   }
+
+  @Override
+  public long transactionIntrinsicGasCost(
+      final Bytes payload, final boolean isContractCreation, final long baselineGas) {
+    final long dynamicIntrinsicGasCost =
+        dynamicIntrinsicGasCost(payload, isContractCreation, baselineGas);
+    final long totalCostFloor =
+        tokensInCallData(payload.size(), zeroBytes(payload)) * TOTAL_COST_FLOOR_PER_TOKEN;
+
+    return clampedAdd(TX_BASE_COST, Math.max(dynamicIntrinsicGasCost, totalCostFloor));
```

Comment:
I believe this is a bug because we shouldn't consider the totalCostFloor as part of the intrinsic gas, rather apply the floor after tx execution.

Note, this code does still work for validating the tx (where the floor is only compared against the intrinsic gas).

According to the latest spec correction, the floor should also be applied _after_ the refund: https://github.com/ethereum/execution-specs/pull/1076/files

In short:
```
gas_used -= refund
gas_used = max(gas_used, floor_cost)
```

where gas_used is the post-execution value rather than intrinsic gas.